enclave/sgx/collateralinfo.c

```diff
@@ -169,6 +170,123 @@ oe_result_t oe_get_sgx_quote_verification_collateral(
     return result;
 }
 
+void oe_prealloc_quote_verification_collateral_args(
+oe_get_sgx_quote_verification_collateral_args_t* buf)
+{
+    /*These numbers are just estimates of buffer sizes*/
+    size_t prealloc_tcb_info_size = 5000;
+    size_t prealloc_tcb_info_issuer_chain_size = 3000;
+    size_t prealloc_pck_crl_size = 600;
+    size_t prealloc_root_ca_crl_size = 600;
+    size_t prealloc_pck_crl_issuer_chain_size = 3000;
+    size_t prealloc_qe_identity_issuer_chain_size = 3000;
+    size_t prealloc_qe_identity_size = 700;
```

Comment:
Can you `#define` these magic numbers at the top of the file? For example `TCBINFO_DEFAULT_SIZE` and the leave a detailed comment of why we're hard-coding these values.

Reply:
Yes, I can do that. My intention was to scope them locally so that they get cleaned up after the function ends. But I do agree they may need to be defined static on the top with some explanation.

Reply:
@jhand2 what do you think about making these variables static inside the function? This is in line with yentsanglee 's suggestion below on line 180. Do you still recommend I #define them? 
I can do both static inside the function and #define them in this file, if needed. Just thought it may be too much of an overhead.

Reply:
I prefer to always `#define` magic numbers to clearly call them out, explain their purpose, and be able to easily change them later. But because they're only used once I don' have a strong preference either way.